### PR TITLE
Unit test model

### PR DESCRIPTION
**Pull request by BritneyS**
## What you did :question:
This PR includes unit tests written for updating models `PlayerModel` and `ChoreModel` via their respective view models `PlayerViewModel` and `ChoreViewModel`. 

## How to test it :microscope:
- Checkout the branch
- Run the tests (in XCode, `cmd + U` is a handy shortcut for this)
- Note that both tests pass

## Any background context you want to provide :question:
N/A

## Screenshots (if applicable) :camera:
N/A

### CODE REVIEW

File: SparkleSpin/PlayerViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+class PlayerViewModel {
+    
+    private let playerModel: PlayerModel
+    var nameString: String? = nil
+    
+    init(playerModel: PlayerModel) {
+        self.playerModel = playerModel
+    }
+    
+    func updateProperties(playerModel: PlayerModel, nameString: String) {
```
Comment (tolkiana) [inline SparkleSpin/PlayerViewModel.swift]: What's the reasoning for passing the Model here again? Looks like you are already passing it in the initializer.

File: SparkleSpin/PlayerViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        setPlayerName(playerModel: playerModel, nameString: nameString)
+    }
+    
+    private func setPlayerName(playerModel: PlayerModel, nameString: String?) {
+        playerModel.name = nameString
+    }
```
Comment (tolkiana) [inline SparkleSpin/PlayerViewModel.swift:22]: What's the purpose of having this private function? I think you could call directly `playerModel.name = nameString` in `updateProperties`  but maybe I'm missing something.

**Review by tolkiana — CHANGES_REQUESTED** [review]
I think this is a good start. I would like to know your answer in the two questions I asked specially for the first one. I think passing the model in two different places could be confusing and lead to bugs.

File: SparkleSpin/PlayerViewModel.swift (hunk 1 above)
Comment (BritneyS) [reply SparkleSpin/PlayerViewModel.swift]: After revisiting this, I can see why this is redundant. I will only pass the `nameString` here.

File: SparkleSpin/PlayerViewModel.swift (hunk 2 above)
Comment (BritneyS) [reply SparkleSpin/PlayerViewModel.swift:22]: I think I did this in anticipation of `PlayerModel` soon having a property of type `Chore`, but there could be a better way of handling that when I get to it. For now, I will call `playerModel.name = nameString` directly.

File: SparkleSpin/ChoreViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        self.choreModel = choreModel
+    }
+    
+    func updateProperties(choreModel: ChoreModel, choreNameString: String) {
+        setPlayerName(choreModel: choreModel, choreNameString: choreNameString)
+    }
+    
+    private func setPlayerName(choreModel: ChoreModel, choreNameString: String?) {
+        choreModel.choreName = choreNameString
+    }
```
Comment (tolkiana) [inline SparkleSpin/ChoreViewModel.swift:22]: Looks like this class needs to be updated too

**Review by tolkiana — CHANGES_REQUESTED** [review]
`ChoreViewModel ` needs to be updated too

**Review by tolkiana — APPROVED** [review]
Good job on these first tests! 🌮
Since you are adding tests to your project I think it would be valuable to explore implementing CI for your project, that way you could immediately see if any tests are broken when opening PRs. 
I think maybe https://travis-ci.org could be an option. I have never used it but I know you can easily integrate with github and it's free as long as your project is open source, this is just one suggestion you could explore other options.

**Pull request merged**